x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/watch/WatchStoreUtilsTests.java

```diff
@@ -0,0 +1,179 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.watcher.watch;
+
+import org.elasticsearch.Version;
+import org.elasticsearch.cluster.metadata.AliasMetadata;
+import org.elasticsearch.cluster.metadata.DataStream;
+import org.elasticsearch.cluster.metadata.DataStreamAlias;
+import org.elasticsearch.cluster.metadata.DataStreamMetadata;
+import org.elasticsearch.cluster.metadata.IndexMetadata;
+import org.elasticsearch.cluster.metadata.Metadata;
+import org.elasticsearch.common.collect.ImmutableOpenMap;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.index.Index;
+import org.elasticsearch.index.IndexMode;
+import org.elasticsearch.test.ESTestCase;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class WatchStoreUtilsTests extends ESTestCase {
+
+    public void testGetConcreteIndexForDataStream() {
+        String dataStreamName = randomAlphaOfLength(20);
+        Metadata.Builder metadataBuilder = Metadata.builder();
+        ImmutableOpenMap.Builder<String, Metadata.Custom> customsBuilder = ImmutableOpenMap.builder();
+        Map<String, DataStream> dataStreams = new HashMap<>();
+        ImmutableOpenMap.Builder<String, IndexMetadata> indexMetadataMapBuilder = ImmutableOpenMap.builder();
+        List<String> indexNames = new ArrayList<>();
+        for (int i = 0; i < randomIntBetween(2, 10); i++) {
+            String indexName = dataStreamName + "_" + i;
+            indexNames.add(indexName);
+            IndexMetadata.Builder indexMetadataBuilder = new IndexMetadata.Builder(indexName);
+            Settings settings = Settings.builder()
+                .put(IndexMetadata.SETTING_PRIORITY, 5)
+                .put(IndexMetadata.INDEX_NUMBER_OF_SHARDS_SETTING.getKey(), 1)
+                .put(IndexMetadata.INDEX_NUMBER_OF_REPLICAS_SETTING.getKey(), 1)
+                .put(IndexMetadata.SETTING_INDEX_VERSION_CREATED.getKey(), Version.CURRENT)
+                .build();
+            indexMetadataBuilder.settings(settings);
+            indexMetadataMapBuilder.put(indexName, indexMetadataBuilder.build());
+        }
+        metadataBuilder.indices(indexMetadataMapBuilder.build());
+        dataStreams.put(
+            dataStreamName,
+            new DataStream(
+                dataStreamName,
+                indexNames.stream().map(indexName -> new Index(indexName, IndexMetadata.INDEX_UUID_NA_VALUE)).collect(Collectors.toList()),
+                randomInt(),
+                Collections.emptyMap(),
+                true,
+                randomBoolean(),
+                true,
+                randomBoolean(),
+                IndexMode.TIME_SERIES
+            )
+        );
```

Comment:
Maybe use a helper method from `DataStreamTestHelper` to create the `DataStream` instance?
(maybe the helper method needs to be modified to accept time series mode or have a dedicated method that creates random tsdb data streams)